src/Infrastructure.EntityFramework/KeyManagement/Repositories/UserSigningKeysRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿#nullable enable
+using AutoMapper;
+using Bit.Core.KeyManagement.Models.Data;
+using Bit.Core.KeyManagement.Repositories;
+using Bit.Core.KeyManagement.UserKey;
+using Bit.Infrastructure.EntityFramework.Repositories;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Bit.Infrastructure.EntityFramework.KeyManagement.Repositories;
+
+public class UserSignatureKeyPairRepository : Repository<Core.KeyManagement.Entities.UserSignatureKeyPair, Models.UserSignatureKeyPair, Guid>, IUserSignatureKeyPairRepository
+{
+    public UserSignatureKeyPairRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper) : base(serviceScopeFactory, mapper, context => context.UserSignatureKeyPair)
+    {
+    }
+
+    public async Task<SignatureKeyPairData?> GetByUserIdAsync(Guid userId)
+    {
+        await using var scope = ServiceScopeFactory.CreateAsyncScope();
+        var dbContext = GetDatabaseContext(scope);
+        var signingKeys = await dbContext.UserSignatureKeyPair.FindAsync(userId);
```

Comment:
```suggestion
       var signingKeys = await dbContext.UserSignatureKeyPair.Where(kp => kp.UserId == userId).FirstOrDefaultAsync();
```

`FindAsync` searches the primary key for the ID provided.